Lab-5/lab5_integration.ipynb

```diff
@@ -0,0 +1,414 @@
+    "> [6] LN-DD2363-part6.pdf, pp. 224-225"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 204,
+   "outputs": [],
+   "source": [
+    "def montecarlounit(f, n):\n",
+    "    \n",
+    "    result = 0\n",
+    "    \n",
+    "    for i in range(n):\n",
+    "        result += f(random.uniform(0,1))\n",
+    "    \n",
+    "    result /= n\n",
+    "    \n",
+    "    return result"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# **Results**"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### 2-point Gauss quadrature over a unit interval  ###\n",
+    "We check that the approximation is close enough to the exact solution."
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 205,
+   "outputs": [
+    {
+     "name": "stdout",
+     "text": [
+      "Test Completed Successfully!\n"
+     ],
+     "output_type": "stream"
+    }
+   ],
+   "source": [
+    "def test_gaussquadunit():\n",
+    "    \n",
+    "    for i in range(1, 1000, 1):\n",
+    "      poly = randpoly()\n",
+    "      apx = gaussquadunit(poly[0])\n",
+    "      assert np.isclose(apx, poly[1])\n",
+    "        \n",
+    "    print(\"Test Completed Successfully!\")\n",
+    "    \n",
+    "test_gaussquadunit()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### 3-point edge midpoint quadrature over a reference triangle  ###\n",
+    "We check that the approximation is close enough to the exact solution."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 206,
+   "outputs": [
+    {
+     "name": "stdout",
+     "text": [
+      "Test Completed Successfully!\n"
+     ],
+     "output_type": "stream"
+    }
+   ],
+   "source": [
+    "def test_gaussquadtriangle():\n",
+    "    \n",
+    "    for i in range(1, 1000, 1):\n",
+    "      tet = randtet()\n",
+    "      apx = gaussquadtriangle(tet[0])\n",
+    "      assert np.isclose(apx, tet[1])\n",
+    "        \n",
+    "    print(\"Test Completed Successfully!\")\n",
+    "    \n",
+    "test_gaussquadtriangle()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Monte Carlo quadrature over a unit interval ###\n",
+    "We check that the error converges towards zero as the number of samples increases. By inspection it is apparent that the convergence rate resembles 1/sqrt(samples)."
```

Comment:
To make it clear that the convergence rate resembles 1/sqrt(n) you need to actually plot 1/sqrt(n) as well. Or a log-log plot. Comparison is a must for you to strengthen your claim.